nominal/core/client.py

```diff
@@ -229,6 +232,38 @@ def search_users(self, exact_match: str | None = None, search_text: str | None =
         query = create_search_users_query(exact_match, search_text)
         return list(self._iter_search_users(query))
 
+    def _iter_search_datasets(self, query: scout_catalog.SearchDatasetsQuery) -> Iterable[Dataset]:
+        for raw_dataset in search_datasets_paginated(self._clients.catalog, self._clients.auth_header, query):
+            yield Dataset._from_conjure(self._clients, raw_dataset)
+
+    def search_datasets(
+        self,
+        exact_match: str | None = None,
```

Comment:
```suggestion
        self,
        *,
        exact_match: str | None = None,
```

Reply:
forces callers to use kw-args to call. especially useful for these APIs that have a lot of string parameters in a row, which is unclear at the callsite otherwise